src/Codec/Picture/Tiff/Types.hs

```diff
@@ -368,15 +368,15 @@ instance BinaryParam B.ByteString (TiffHeader, [[ImageFileDirectory]]) where
 data TiffSampleFormat
   = TiffSampleUint
   | TiffSampleInt
-  | TiffSampleDouble
+  | TiffSampleFloat
```

Comment:
The name `TiffSampleDouble` seems quite confusing; at least it had me scratching my head for a minute or two. What's more ,`unpack` as it's currently written will fail to process TIFFs with 64-bit floating point samples; perhaps we should support that too, since it doesn't seem to be excluded by the standard.